content/automate/ManageIQ/Transformation/Infrastructure/VM/Common.class/__methods__/restorevmattributes.rb

```diff
@@ -5,31 +5,57 @@ module Infrastructure
         module VM
           module Common
             class RestoreVmAttributes
+              IDENTITY_ITEMS = %w(service tags custom_attributes).freeze
+
               def initialize(handle = $evm)
                 @handle = handle
               end
 
-              def main
-                task = @handle.root['service_template_transformation_plan_task']
-                source_vm = task.source
-                destination_vm = @handle.vmdb(:vm).find_by(:id => task.get_option(:destination_vm_id))
-                @handle.log(:info, "VM: #{destination_vm.name} [#{destination_vm.id}]")
+              def log_and_raise(message)
+                @handle.log(:error, message)
+                raise "ERROR - #{message}"
+              end
+
+              def task
+                @task ||= @handle.root["service_template_transformation_plan_task"].tap do |task|
+                  log_and_raise('task object is not passed in') if task.nil?
+                end
+              end
+
+              def source_vm
+                @vm ||= task.source.tap do |vm|
+                  log_and_raise('task.source is not set') if vm.nil?
+                end
+              end
 
-                # Reconnect destination VM to service
+              def destination_vm
+                log_and_raise("task has no ':destination_vm_id' option") if task.get_option(:destination_vm_id).blank?
```

Comment:
@fdupont-redhat 
log_and_raise('destination vm id is blank')

and then
log_and_raise('destination vm not found after migration in the database')